api/openapi-spec/swagger.json

```diff
@@ -3352,6 +3352,10 @@
      "cores": {
       "description": "Cores specifies the number of cores inside the vmi.\nMust be a value greater or equal 1.",
       "type": "integer"
+     },
+     "model": {
+      "description": "Model specifies the CPU model inside the VMI.\n+optional",
+      "type": "string"
```

Comment:
Just like the interface models we should validate this against a list of CPU models (+ host-model).

Do we have a way to validate this value against a set of values?

Reply:
@fabiand We can copy available models from https://github.com/wiedi/libvirt/blob/master/src/cpu/cpu_map.xml, but do we want to do it? It likes to validate CPU features, I believe if someone specifies model he at least must know what he is doing.
But if you prefer strict validation, I can add it.

Reply:
First: I'll not block.
But it's something we need to solve, because: If in this case the user puts a weird value in, then it needs som digging (aka look at a few entities) to find the root cause for the VM to not be launched.

I actually wonder,  @booxter , did you settle on an approach to validate nic models? (the problem is similar)

Reply:
agreed with fabiand it's not really required right now, but once we add it let's please use an API, not a copy of internal .xml

Reply:
@fabiand for network models, we have a closed set of possible values that is more narrow than what qemu would actually allow.

In addition to concerns about debugging complexity, another thing to consider is that allowing *any* CPU model / vendor may expose us to more CVEs specific to a particular hardware model / vendor if they are ever found in underlying components (libvirt / qemu). Controlling the list potentially reduces the attack area.

Reply:
@michalskrivanek I do not copy libvirt XML with supported CPU types, so we do not have any validation for now.
I also do not see the possibility to use API to fetch available CPU models from libvirt, because we do not run libvirt inside a node. We can fetch capabilities from `virt-launcher` pod, but it will happen after validation stage, so it does not really help us.

Reply:
@cynepco3hahue having "smart" logic is a next step. if we do initial validation, then it should just be a whitelist. Nothing smarter.

Let's do smaller steps.

Reply:
Let's make it short: No semantic validation here. 

Can be up next,let's close this gap for now.

Reply:
> Can be up next

yep, that work needs to be started asap (with virtuned, ideally). it's not easy, but we'll be running into similar validation issues again and again.